README.md

```diff
@@ -424,7 +426,49 @@ Assumes database resolution is done by adding prefixes. If database resolution i
      where h.date = '2016-05-13'
        and h.hour = 1
     ;
-    
+
+## Database Name Mapping
+NOTE: mapping names adds an extra layer of abstraction and we advise to use this as a temporary migration solution only. It becomes harder to debug where a virtual (remapped) table actually is coming from.
+
+Waggle Dance allows one to refer to databases by names that are different to what they are defined in the Hive metastore via a `database-name-mapping` configuration. This feature can be useful when migrating data from existing databases into different environments.
+To clarify how this feature could be used, below is an example use case:
+As part of a data migration we have decided that we want to store all Hive tables related to hotel bookings in a database called `booking`. However we have legacy data lakes that have tables in this domain but they are stored in databased with names that don't match this new standard, e.g. 'datawarehouse' or 'hotel_booking'. When federating queries to those legacy data lakes it helps data discovery if the database names are standardized to new name. Renaming a database in Hive is easy but updating all the relevant ETL scripts and jobs might not be, so to make migration easier you can configure Waggle Dance to map the database names so both the new and old names can be used. At some point in the future when all the references to the old names are updated this mapping can then be removed.
```

Comment:
Now that we've discussed this more I see my use case below isn't correct. This feature is more about allowing you to refer to things by the name they _will_ have in the future so you can future proof your ETL and scripts etc. and then migrate them later so you build up less technical debt while a migration is in progress. @barnharts4 if I'm correct would you mind having a go at using your wordsmithery and writing a better use case?

Reply:
Actually, I was under the impression that the use-case *was* for referring to databases by their old name for federation so as to allow a smooth migration for legacy federation customers.  Then, when the customer has had time to migrate all ETL processes and scripts to the new name, the old name in the Waggledance config could be deprecated.  That is certainly the use-case we had in mind for our internal EGDP migration.

Reply:
so you would want both names? the federated one and as an extra a renamed one?

Reply:
Yes, that would be what is most helpful for doing a migration of database names with minimal disruption to legacy customers.

Reply:
ok will revisit this, thanks

Reply:
@barnharts4 @massdosage Ok updated the PR please have a look if you got time and let me if this makes more sense :) Thanks!